sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -70,11 +70,25 @@ public virtual RecordEntry FindMatch(Request request, IList<RecordEntry> entries
             int bestScore = int.MaxValue;
             RecordEntry bestScoreEntry = null;
 
+            bool isTrack1Record = entries.Any(item => !string.IsNullOrEmpty(item.EncodedRequestUri));
+
             foreach (RecordEntry entry in entries)
             {
                 int score = 0;
 
-                if (!AreUrisSame(entry.RequestUri, uri))
+                var recordRequestUri = entry.RequestUri;
+                if (isTrack1Record)
+                {
+                    //there's no domain name for request uri in track 1 record, so add it from reqeust uri
+                    int len = 8; //length of "https://"
+                    int domainEndingIndex = uri.IndexOf('/', len);
+                    if (domainEndingIndex > 0)
+                    {
+                        recordRequestUri = uri.Substring(0, domainEndingIndex) + recordRequestUri;
```

Comment:
Let's move this logic into RecordEntry.Deserialize this would allow us to drop the additional RequestUri property.

Reply:
There's no domain name for request uri in track 1 record, so I get domain name from **real** request uri and add domain name to RequestUri of track 1 for comparison, the problem is we don't know real request uri during de-serializing. Please let me know if I have some misunderstanding.

The real request uri: https://xxx.com/subscriptions/xxxx
RequestUri in track 1: /subscriptions/xxxxx
RequestUri in track 2: https://xxx.com/subscriptions/xxxx